samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -16,85 +16,327 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.apache.samza.operators.windows;
 
+import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.operators.data.MessageEnvelope;
+import org.apache.samza.operators.triggers.TimeTrigger;
+import org.apache.samza.operators.triggers.Trigger;
+import org.apache.samza.operators.triggers.Triggers;
 
+import java.time.Duration;
 import java.util.Collection;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
-
 /**
- * This class defines a collection of {@link Window} functions. The public classes and methods here are intended to be
- * used by the user (i.e. programmers) to create {@link Window} function directly.
+ * Factory methods for creating a {@link Window}. Provides APIs for specifying different types of {@link Window}s.
+ * A {@link Window} can have a specified keyFunction that is used to group {@link MessageEnvelope}s in the {@link Window}.
  *
+ * <p> The following types of {@link Window}s are supported.
+ * <ul>
+ *  <li>Tumbling Windows: A tumbling window defines a series of non-overlapping, fixed size, contiguous time intervals.
+ *  <li>Session Windows: A session window groups a {@link org.apache.samza.operators.MessageStream} into sessions.
+ *  A <i>session</i> captures some period of activity over a {@link org.apache.samza.operators.MessageStream}.
+ *  The boundary for a session is defined by a {@code sessionGap}. All {@link MessageEnvelope}s that that arrive within
+ *  the gap are grouped into the same session.
+ *  <li>Global Windows: A global window defines a single infinite window over the entire {@link org.apache.samza.operators.MessageStream}.
+ *  An early trigger must be specified when defining a global window.
+ * </ul>
  */
+@InterfaceStability.Unstable
 public final class Windows {
 
+  private Windows() {
+
+  }
+
   /**
-   * private constructor to prevent instantiation
+   * Creates a {@link Window} that groups incoming {@link MessageEnvelope}s using the provided keyFn into
+   * fixed-size, non-overlapping processing time based windows and applies the provided aggregate function to them.
```

Comment:
"groups into ... processing time windows", should we introduce a new term for a row of window panes?

"applies the provided fold function"

"groups incoming messages into ... using the provided keyFn"